### PR TITLE
[Workflow] Move code from ValidateWorkflowsPass to the FrameworkExtension

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |
| License       | MIT
| Doc PR        |

---

Just some cleaning. See
https://github.com/symfony/symfony/pull/29146#issuecomment-44418654

### CODE REVIEW

File: src/Symfony/Component/Workflow/DependencyInjection/ValidateWorkflowsPass.php [hunk 1]
```diff
@@ -1,64 +0,0 @@
...
-use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
-use Symfony\Component\DependencyInjection\ContainerBuilder;
-use Symfony\Component\DependencyInjection\Exception\RuntimeException;
-use Symfony\Component\Workflow\Validator\StateMachineValidator;
-use Symfony\Component\Workflow\Validator\WorkflowValidator;
-
-/**
- * @author Tobias Nyholm <tobias.nyholm@gmail.com>
- */
-class ValidateWorkflowsPass implements CompilerPassInterface
```
Comment (xabbuh) [inline src/Symfony/Component/Workflow/DependencyInjection/ValidateWorkflowsPass.php]: Did we never register the pass inside the FrameworkBundle? Also, removing the class here is a BC break for anyone using it in a standalone setup.

**Review by xabbuh — DISMISSED** [review]

File: src/Symfony/Component/Workflow/DependencyInjection/ValidateWorkflowsPass.php (hunk 1 above)
Comment (lyrixx) [reply src/Symfony/Component/Workflow/DependencyInjection/ValidateWorkflowsPass.php]: Indeed, I forgot to remove it from the framework bundle.
I'm not sure if we should consider it as a BC break for the reason I explained in https://github.com/symfony/symfony/pull/29146#issuecomment-444194286

**Comment by stof** [comment]
The compiler pass should be deprecated, not removed. It don't think making such BC break is actually worth it compared to the deprecation path.

**Comment by lyrixx** [comment]
Okay, Il will deprecate it. But I should keep the code there ?

**Comment by lyrixx** [comment]
Hi. I rebased and finished this PR. I put back the ValidateWorkflowsPass but I deprecate it.
I think it's OK now

**Comment by lyrixx** [comment]
@stof @xabbuh Are you OK with this one? Could I merge it?

**Pull request merged**